tests/test_services/test_analytics_service.py

```diff
@@ -39,6 +50,83 @@ def mocked_global_parameters_service(matcha_testing_directory, uuid_for_testing)
         yield GlobalParameters()
 
 
+def test_get_state_uuid_with_state(uuid_for_testing):
+    """Test the _get_state_uuid private function where state exists.
+
+    Args:
+        uuid_for_testing (uuid.UUID): a fixed valid UUID4 value.
+    """
+    with patch(MATCHA_STATE_SERVICE_FUNCTION_STUB) as mock_matcha_state_service:
+        mock_matcha_state_service.return_value.state_exists.return_value = True
+        mock_component = (
+            mock_matcha_state_service.return_value.get_component.return_value
+        )
+        mock_component.find_property.return_value.value = uuid_for_testing
+        result = _get_state_uuid()
+
+    assert result.value == uuid_for_testing
+
+
+def test_execute_analytics_event():
+    """Test the _execute_analytics_event private function."""
```

Comment:
NIT: Could require a little more detail about the expected behaviour of the test, e.g. "Test the _execute_analytics_event private function calls the function passed to it as a parameter."